src/util/hooks.js

```diff
@@ -0,0 +1,19 @@
+/* @flow */
+
+export function callThroughHook<T>(type: string, fn: () => T): T {
```

Comment:
Let's name this `triggerHook` or `callHook` and use an enum or `Symbol`s for hook names.

Reply:
I prefer `callThroughHook` because it underlines that the hook has the responsibility of actually calling the callback.

I improved the Flow typing of the `type` parameter to be an enumeration that only accepts valid field. Symbols aren't possible because the hooks will be setup before Yarn is even required.